sematic/db/queries.py

```diff
@@ -64,7 +64,12 @@ def get_artifact(artifact_id: str) -> Artifact:
         return session.query(Artifact).filter(Artifact.id == artifact_id).one()
 
 
-def get_users(user_ids: List[str]) -> List[User]:
+def get_users() -> List[User]:
+    with db().get_session() as session:
+        return session.query(User).order_by(User.first_name).all()
```

Comment:
I guess the ideal logic will be `SELECT DISTINCT user_id FROM runs JOIN users on users.id=runs.user_id`

Because we are creating a filter for runs. There might be cases in which some users are not associated with runs (eg. notes only users).

Reply:
Even more than that, shouldn't it be users that have submitted runs for the currently-viewed pipeline?

We either need a highly specific endpoint, or a highly flexible way of specifying filters. I'd go with the first version until we have more use cases.

Reply:
Let's keep this dumb and list all users. Otherwise we would also need to restrict run statuses in the run status filter  to only available statuses for a given list of runs.

We can do facets (Elastic Search term) when we use a dedicated search engine.

Reply:
I fear that too many users might even render the UI unusable. Is that concerning or not?

Reply:
@chance-an this is not likely to happen for a long time. Even with SaaS, users would only see people from their org, which I expect should be under 20.